### PR TITLE
[flutter_conductor] begin migrating //flutter/dev/tools to null-safety

### PR DESCRIPTION
**Pull request by christopherfujino**
Several of the conductor files are blocked from migrating because of transitive dependencies that are still pinned to pre-null-safety. Of the other legacy files in this folder, I migrated a few easy ones, and left the rest for a future migration.

### CODE REVIEW

File: dev/tools/lib/repository.dart [hunk 1]
```diff
@@ -84,31 +83,31 @@ abstract class Repository {
...
   final bool localUpstream;
 
-  Directory _checkoutDirectory;
+  Directory? _checkoutDirectory;
 
   /// Directory for the repository checkout.
   ///
   /// Since cloning a repository takes a long time, we do not ensure it is
   /// cloned on the filesystem until this getter is accessed.
   Directory get checkoutDirectory {
```
Comment (jonahwilliams) [inline dev/tools/lib/repository.dart:92]: You could use `late final` to set this up automagically, assuming some light wrapper

```dart
late final Directory checkoutDirectory = lazilyInitialize(parentDirectory.childDirectory(name));
```

You might need to tweak `lazilyInitialize` to return the original directory

**Review by jonahwilliams — APPROVED** [review]
LGTM with a small style nit

File: dev/tools/lib/repository.dart (hunk 1 above)
Comment (christopherfujino) [reply dev/tools/lib/repository.dart:92]: good point, i'll try that. i also need to fix the tests.

Comment (christopherfujino) [reply dev/tools/lib/repository.dart:92]: Migrating to late final is breaking my test, I'll follow up with a PR isolating that change.

**Pull request merged**